.github/workflows/test-python.yaml

```diff
@@ -0,0 +1,133 @@
+name: Test Python
+on:
+  workflow_call:
+    inputs:
+      unit-platforms:
+        required: false
+        type: string
+        default: '["ubuntu-22.04", "ubuntu-24.04", "windows-latest", "macos-latest"]'
+        description: |
+          The platforms to run unit tests on, as a JSON array.
+      unit-python-versions:
+        required: false
+        type: string
+        default: '["3.10", "3.11", "3.12", "3.13"]'
+        description: |
+          The python versions to run unit tests on, as a JSON array.
+      lowest-python-version:
+        required: false
+        type: string
+        description: |
+          The Python version to run when using "lowest" resolution for unit tests.
+      integration-platforms:
+        type: string
+        default: '["ubuntu-latest", "windows-latest", "macos-latest"]'
+        description: |
+          The platforms to run integration tests on, as a JSON array.
+      integration-python-versions:
+        required: false
+        type: string
+        default: '["3.10"]'
+        description: |
+          The python versions to run integration tests on, as a JSON array.
+
+jobs:
+  unit:
+    strategy:
+      matrix:
+        platform: ${{ fromJson(inputs.unit-platforms) }}
+    runs-on: ${{ matrix.platform }}
+    env:
+      UV_PYTHON: ${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up uv with caching
+        id: setup-uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+          cache-suffix: ${{ matrix.platform }}
+      - name: Set up tests
+        shell: bash
+        run: |
+          for python_version in $(echo '${{ inputs.unit-python-versions }}' | jq -r .[] | tr '\n' ' '); do
+            python_dirname=$(echo ${{ runner.temp }} | tr '\\' /)/venv_$(echo $python_version | tr . _)
+            make setup-tests UV_PROJECT_ENVIRONMENT="${python_dirname}"
+          done
+      - name: Run tests
+        shell: bash
+        run: |
+          for python_version in $(echo '${{ inputs.unit-python-versions }}' | jq -r .[] | tr '\n' ' '); do
+            echo "::group::Python ${python_version}"
+            python_dirname=$(echo ${{ runner.temp }} | tr '\\' /)/venv_$(echo $python_version | tr . _)
+            make coverage UV_PROJECT_ENVIRONMENT="${python_dirname}"
+            echo "::endgroup::"
+          done
+      - name: Upload coverage
+        if: ${{ inputs.target == 'coverage' }}
```

Comment:
do you have an example that matches this? As far as I can tell all unit-test runs had coverage disabled on the Starbase PR:
![image](https://github.com/user-attachments/assets/7fbd7f4a-4bce-4da1-94bb-4f748be7794d)